src/test/controller_mapping_validation_test.cpp

```diff
@@ -163,35 +159,69 @@ bool lintMappingInfo(const MappingInfo& mapping) {
     return result;
 }
 
-TEST_F(LegacyControllerMappingValidationTest, MidiMappingsValid) {
-    foreach (const MappingInfo& mapping,
-            m_pEnumerator->getMappingsByExtension(MIDI_MAPPING_EXTENSION)) {
-        qDebug() << "Validating " << mapping.getPath();
-        std::string errorDescription = "Error while validating " + mapping.getPath().toStdString();
-        EXPECT_TRUE(mapping.isValid()) << errorDescription;
-        EXPECT_TRUE(lintMappingInfo(mapping)) << errorDescription;
-        EXPECT_TRUE(testLoadMapping(mapping)) << errorDescription;
-    }
+// Create a mapping enumerator for the test, which will be used to get the mappings
+// for the test cases
+std::shared_ptr<LegacyControllerMappingList> createMappingEnumerator() {
+    return std::make_shared<LegacyControllerMappingList>();
 }
 
-TEST_F(LegacyControllerMappingValidationTest, HidMappingsValid) {
-    foreach (const MappingInfo& mapping,
-            m_pEnumerator->getMappingsByExtension(HID_MAPPING_EXTENSION)) {
-        qDebug() << "Validating" << mapping.getPath();
-        std::string errorDescription = "Error while validating " + mapping.getPath().toStdString();
-        EXPECT_TRUE(mapping.isValid()) << errorDescription;
-        EXPECT_TRUE(lintMappingInfo(mapping)) << errorDescription;
-        EXPECT_TRUE(testLoadMapping(mapping)) << errorDescription;
-    }
+// Inhibit the output of the mapping info to avoid spamming the console
+std::ostream& operator<<(std::ostream& os, const MappingInfo& mapping) {
+    Q_UNUSED(mapping);
+    os << "<MappingInfo>";
+    return os;
 }
 
-TEST_F(LegacyControllerMappingValidationTest, BulkMappingsValid) {
-    foreach (const MappingInfo& mapping,
-            m_pEnumerator->getMappingsByExtension(BULK_MAPPING_EXTENSION)) {
-        qDebug() << "Validating" << mapping.getPath();
-        std::string errorDescription = "Error while validating " + mapping.getPath().toStdString();
-        EXPECT_TRUE(mapping.isValid()) << errorDescription;
-        EXPECT_TRUE(lintMappingInfo(mapping)) << errorDescription;
-        EXPECT_TRUE(testLoadMapping(mapping)) << errorDescription;
+// Print the mapping name instead with underscores instead of spaces and special characters
+class PrintMappingName {
+  public:
+    template<class ParamType>
+    std::string operator()(const ::testing::TestParamInfo<ParamType>& info) const {
+        // ParamType is MappingInfo
+        QString nameCopy = info.param.getName();
+        static const QRegularExpression regex("[^\\w\\s]");
+        nameCopy.replace(regex, "_");
+        std::string testName = nameCopy.replace(" ", "_").toStdString();
+        return testName;
     }
+};
+
+auto pEnumerator = createMappingEnumerator();
```

Comment:
this relies on running before main. I don't know how safe that is considering the heavy Qt classes involved. It would be better if this was run in `main()`, but before the gtest init code. Is that feasible?